keyring/backend.py

```diff
@@ -250,7 +250,7 @@ def _query(self, service, username=None, **base):
                 scheme['username']: username,
                 scheme['service']: service,
             }
-            if username
+            if username is not None
```

Comment:
It's possible some users are relying on `username = ''` as a non-value for username. We can address that separately if needed.